explore/src/main/scala/explore/observationtree/obsTree.scala

```diff
@@ -8,40 +8,59 @@ import cats.Order.*
 import cats.derived.*
 import cats.syntax.all.*
 import eu.timepit.refined.cats.*
+import eu.timepit.refined.types.numeric.NonNegShort
 import explore.data.KeyedIndexedList
-import explore.model.GroupList
-import explore.model.GroupObs
-import explore.model.Grouping
-import explore.model.GroupingElement
-import explore.model.ObsSummary
+import explore.data.tree.KeyedIndexedTree
+import explore.data.tree.{Node => IndexNode}
+import explore.data.tree.{Tree => IndexTree}
+import explore.model.*
+import explore.model.syntax.all.*
+import lucuma.core.model.Group
 import lucuma.core.model.Observation
 import lucuma.react.primereact.Tree
 import lucuma.react.primereact.Tree.Node
+import monocle.Iso
 
 enum ObsNode derives Eq:
```

Comment:
Why an `enum`? It seems this would be more an `ADT`

Reply:
Aren't enums with params basically ADT's?

Reply:
Yes